tests/unit/amo/components/TestRating.js

```diff
@@ -69,411 +95,342 @@ describe(__filename, () => {
-      );
+      const span = screen.getByText(`Rate this add-on ${rating} out of 5`);
+      expect(span).toHaveClass('visually-hidden');
     });
   });
 
   it('renders an accessible description for null stars', () => {
-    const root = renderWithEmptyRating();
+    renderWithEmptyRating();
 
     [1, 2, 3, 4, 5].forEach((rating) => {
-      expect(getStar({ root, rating })).toHaveProp(
-        'title',
-        `Rate this add-on ${rating} out of 5`,
-      );
+      expect(
+        screen.getByTitle(`Rate this add-on ${rating} out of 5`),
+      ).toBeInTheDocument();
     });
   });
 
   it('renders an accessible description for 0 stars', () => {
-    const root = render({ rating: 0 });
+    render({ rating: 0 });
 
     [1, 2, 3, 4, 5].forEach((rating) => {
-      expect(getStar({ root, rating })).toHaveProp(
-        'title',
-        `Rate this add-on ${rating} out of 5`,
-      );
+      expect(
+        screen.getByTitle(`Rate this add-on ${rating} out of 5`),
+      ).toBeInTheDocument();
     });
   });
 
   it('renders an appropriate title when updating a rating', () => {
     const userRating = 3;
-    const root = render({ rating: userRating });
+    render({ rating: userRating });
 
     [1, 2, 4, 5].forEach((rating) => {
-      expect(getStar({ root, rating })).toHaveProp(
-        'title',
-        `Update your rating to ${rating} out of 5`,
-      );
+      expect(
+        screen.getByTitle(`Update your rating to ${rating} out of 5`),
+      ).toBeInTheDocument();
     });
 
-    expect(getStar({ root, rating: userRating })).toHaveProp(
-      'title',
-      `Rated ${userRating} out of 5`,
-    );
+    expect(
+      screen.getByTitle(`Rated ${userRating} out of 5`),
+    ).toBeInTheDocument();
   });
 
   it('prevents form submission when selecting a rating', () => {
-    const root = renderWithRating({ onSelectRating: sinon.stub() });
+    renderWithEmptyRating({ onSelectRating: jest.fn() });
+    const star = getStarButton({ rating: 1 });
 
-    const fakeEvent = createFakeEvent();
-    const star = getStar({ root, rating: 4 });
-    star.simulate('click', fakeEvent);
+    const clickEvent = createEvent.click(star);
+    const preventDefaultWatcher = jest.spyOn(clickEvent, 'preventDefault');
+    const stopPropagationWatcher = jest.spyOn(clickEvent, 'stopPropagation');
 
-    sinon.assert.called(fakeEvent.preventDefault);
-    sinon.assert.called(fakeEvent.stopPropagation);
+    fireEvent(star, clickEvent);
+    expect(preventDefaultWatcher).toHaveBeenCalled();
+    expect(stopPropagationWatcher).toHaveBeenCalled();
   });
 
-  it('requires a valid onSelectRating callback', () => {
-    const root = renderWithRating({ onSelectRating: null });
-    const star = getStar({ root, rating: 4 });
-
-    expect(() => star.simulate('click', createFakeEvent())).toThrowError(
-      /onSelectRating was empty/,
-    );
-  });
+  // TODO: I can't seem to figure out how to test this. expect.toThrow
```

Comment:
This one is annoying. I'm not sure why it's not working. The `except(()=>{}).toThrow()` logic *should* work. The exception is thrown, but the `toThrow()` fails claiming nothing was thrown. I also tried adding a `try/catch` into the test around the call to `userEvent.click(star)` but the `catch` never gets fired. I'm guessing that the error is being caught somewhere that I don't know about, which would explain both of those behaviours. Maybe it's being caught by `userEvent.click()` which is part of react-testing-library.

I'm not sure we absolutely need this test. We could just change the `if` statement in `Rating` to an `invariant`, and we generally don't test for invariants, but I also would like to figure out exactly what the problem is here.

@willdurand Pinging you to see if anything obvious comes to mind.

Reply:
I think this is https://github.com/testing-library/react-testing-library/issues/624 so I guess there is nothing we can do.